contracts/secret-storage/SecretStore.sol

```diff
@@ -516,23 +516,26 @@ contract SecretStore is
 
     function _slashEnclave(
         address _enclaveAddress,
-        uint256 _sizeLimit,
-        address _recipient,
-        uint256 _secretId
-    ) internal returns (uint256) {
-        uint256 totalComp = (secretStorage[_enclaveAddress].stakeAmount * SLASH_PERCENT_IN_BIPS) / SLASH_MAX_BIPS;
-        secretStorage[_enclaveAddress].stakeAmount -= totalComp;
+        uint256 _endTimestamp,
+        uint256 _markAliveTimeout,
+        address _recipient
+    ) internal {
+        uint256 lastAliveTimestamp = secretStorage[_enclaveAddress].lastAliveTimestamp;
+        uint256 deadTimestamp = secretStorage[_enclaveAddress].deadTimestamp;
+        uint256 lastCheckTimestamp = (lastAliveTimestamp > deadTimestamp) ? lastAliveTimestamp : deadTimestamp;
+        uint256 missedEpochsCount = (_endTimestamp - lastCheckTimestamp) / _markAliveTimeout;
 
-        STAKING_TOKEN.safeTransfer(_recipient, totalComp);
+        if(missedEpochsCount > 0) {
+            uint256 totalComp = (missedEpochsCount * secretStorage[_enclaveAddress].stakeAmount * SLASH_PERCENT_IN_BIPS) / SLASH_MAX_BIPS;
```

Comment:
there is an edge case, totalComp can go higher than stakeAmount. Resolve it like sum of GP (geometric progression). Think, slashing it n times, each time by x% of resultant amount.